### PR TITLE
ddl: throw unsupported warning when alter table with/without validation

### PR DESCRIPTION
**Pull request by cncal**
### What problem does this PR solve?

Fix [parser issue #606](https://github.com/pingcap/parser/issues/606)

### What is changed and how it works?

```
mysql> use test
Database changed
mysql> create table t1 (c1 int, c2 int as (c1 + 1));
Query OK, 0 rows affected (0.03 sec)

mysql> alter table t1 with validation;
Query OK, 0 rows affected, 1 warning (0.00 sec)

mysql> show warnings;
+---------+------+------------------------------------------------------+
| Level   | Code | Message                                              |
+---------+------+------------------------------------------------------+
| Warning | 8200 | ALTER TABLE WITH VALIDATION is currently unsupported |
+---------+------+------------------------------------------------------+
1 row in set (0.00 sec)

mysql> alter table t1 without validation;
Query OK, 0 rows affected, 1 warning (0.00 sec)

mysql> show warnings;
+---------+------+---------------------------------------------------------+
| Level   | Code | Message                                                 |
+---------+------+---------------------------------------------------------+
| Warning | 8200 | ALTER TABLE WITHOUT VALIDATION is currently unsupported |
+---------+------+---------------------------------------------------------+
1 row in set (0.00 sec)

```

### Check List

Tests

- Unit test

### Release note

- No release note

### CODE REVIEW

File: ddl/db_test.go [hunk 1]
```diff
@@ -4548,6 +4548,30 @@ func (s *testSerialDBSuite) TestDDLJobErrorCount(c *C) {
...
+}
+
+func (s *testDBSuite2) TestAlterTableWithoutValidation(c *C) {
+	s.tk = testkit.NewTestKit(c, s.store)
+	s.tk.MustExec("use test")
+	s.tk.MustExec("drop table if exists t1")
+	defer s.tk.MustExec("drop table if exists t1")
+
+	s.tk.MustExec("create table t1 (c1 int, c2 int as (c1 + 1));")
+	s.tk.MustExec("alter table t1 without validation")
```
Comment (zimulala) [inline ddl/db_test.go:4570]: Could we put this test to `TestAlterTableWithValidation`? Then we can reduce the time to run the test.

Comment (cncal) [reply ddl/db_test.go:4570]: All right.

**Comment by cncal** [comment]
@zimulala @tangenta Conflicts resolved.

**Pull request merged**